lib/repl.js

```diff
@@ -533,14 +553,30 @@ function REPLServer(prompt,
     }
   };
 
+  self.displayWelcomeMessage = !!options.displayWelcomeMessage &&
+                                self.terminal;
+  if (self.displayWelcomeMessage) {
+    const version = process.version;
+    const jsEngine = process.jsEngine || 'v8';
+    const jsEngineVersion = process.versions[jsEngine];
+    const docURL = `https://nodejs.org/dist/${version}/docs/api/repl.html`;
+
+    const welcome = `Welcome to Node.js ${version} (${jsEngine} VM,` +
+                    ` ${jsEngineVersion})\nType ^M or enter to execute,` +
```

Comment:
I think `Node.js ${version} (${jsEngine} ${jsEngineVersion})` should suffice; the "VM" seems a bit extraneous.